### ISSUE TITLE
Testing validation and submit button status

### ISSUE THREAD

**Issue by ChrisMacpherson** [tylercollier/redux-form-test#10]
Hi, 

I've found your repo here very useful. It's actually brought me back from the edge a few times now so thanks!

I am looking to create some tests that say something like:

- When I change the value of text field to empty, then I get an error message saying 'There is an error'.

- When I change the value of text field to empty, then the submit button is disabled.

I'm just wondering what would be the best way to go about this as I can't seem to get much working. 

I am using field level validation (https://redux-form.com/7.4.2/examples/fieldlevelvalidation/).

It's so frustrating using this package as there are no details about how to test things and it seems so over-complicated, unfortunately I didn't choose to use it and am stuck working with it.

**Comment by ChrisMacpherson** [comment]
Okay so I still have hair left and I got something working :) Maybe not perfect yet but my test passes. If anyone has any comments on how to improve my code then feel free to comment.

In the end I had to manually mark the field as touched. I have another test with a <select> where I do not have to do this though so am a little confused. 
```
it('should show an error when value is changed to empty', () => {
      // Given
      const form = formContainer.find('form');

      // When
      form.find('input#myfield').simulate('change', { target: { value: '' } });
      form.find('input#myfield').props().touched = true; // This is required

      // Then
      expect(formContainer.find('.alert-danger').text()).toEqual('required');
});
```

**Comment by ChrisMacpherson** [comment]
Okay so I still have hair left and I got something working :) Maybe not perfect yet but my test passes. If anyone has any comments on how to improve my code then feel free to comment.

In the end I had to manually mark the field as touched. I have another test with a `<select>` where I do not have to do this though so am a little confused.


```javascript
    it('should show an error when value is changed to empty', () => {
      // Given
      const form = formContainer.find('form');

      // When
      form.find('input#myfield').simulate('change', { target: { value: '' } });
      form.find('input#myfield').props().touched = true; // This is required

      // Then
      expect(formContainer.find('.alert-danger').text()).toEqual('required');
    });
```

**Comment by ChrisMacpherson** [comment]
Actually worked out on subsequent tests that the '.touched' requirement is not actually what was helping. 

It was working because after I simulated an event, I was 're-finding' the node again from the wrapper which was updated. It seems there have been changes between Enzyme v2 and v3 regarding updating components.

For more info read here: https://github.com/airbnb/enzyme/blob/master/docs/guides/migration-from-2-to-3.md#for-mount-updates-are-sometimes-required-when-they-werent-before
